storagenode/storagenodedb/database.go

```diff
@@ -290,24 +332,57 @@ func (db *DB) RoutingTable() (kdb, ndb, adb storage.KeyValueStore) {
 
 // RawDatabases are required for testing purposes
 func (db *DB) RawDatabases() map[string]SQLDB {
-	return map[string]SQLDB{
-		BandwidthDBName:       db.bandwidthDB,
-		OrdersDBName:          db.ordersDB,
-		PieceExpirationDBName: db.pieceExpirationDB,
-		PieceSpaceUsedDBName:  db.pieceSpaceUsedDB,
-		ReputationDBName:      db.reputationDB,
-		StorageUsageDBName:    db.storageUsageDB,
-		UsedSerialsDBName:     db.usedSerialsDB,
-		PieceInfoDBName:       db.v0PieceInfoDB,
-		DeprecatedInfoDBName:  db.deprecatedInfoDB,
-		SatellitesDBName:      db.satellitesDB,
+	return db.sqlDatabases
+}
+
+// migrateToDB is a helper method that performs the migration from the
+// deprecatedInfoDB to the specified new db. It first closes and deletes any
+// existing database to guarantee idempotence. After migration it also closes
+// and re-opens the new database to allow the system to recover used disk space.
+func (db *DB) migrateToDB(ctx context.Context, dbName string, tablesToKeep ...string) error {
```

Comment:
@egonelbre see this function. Here we close and reopen the underlying sql.DB several times to guarantee idempotency and to free disk space. We do this by calling the `closeDatabase` and `openDatabase` functions.

Essentially because we need to be able to close and reopen we need to swap out the underlying sql DB for, say, the orders db.

Reply:
Ah, yeah... I was thinking that maybe some other method in the database implementation creates a local tx for multiple calls.